icefall/checkpoint.py

```diff
@@ -151,3 +154,99 @@ def average_checkpoints(
             avg[k] //= n
 
     return avg
+
+
+def save_checkpoint_with_global_batch_idx(
```

Comment:
I think that you should also pass `sampler` here and save its state dict.

Reply:
Yes, you are right.